src/Sylius/Behat/Page/Shop/Product/ShowPage.php

```diff
@@ -226,9 +226,29 @@ public function isMainImageDisplayed()
     /**
      * {@inheritdoc}
      */
+    public function hasAssociation($productAssociationName)
+    {
+        return null !== $this->getElement('association', ['%association-name%' => $productAssociationName]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasProductInAssociation($productName, $productAssociationName)
+    {
+        $associationHeader = $this->getElement('association', ['%association-name%' => $productAssociationName]);
+        $associations = $associationHeader->getParent()->find('css', '.four');
```

Comment:
`Assert::notNull($associations)` to prevent from calling `->find(` on `null` object